### DISCUSSION TITLE
What's the difference between `dotnet watch` and `Hot Reload in Visual Studio`?

### DISCUSSION THREAD

**Discussion by LuohuaRain** [dotnet/aspnetcore#50936] [Q&A]
Interestingly, in my own experimentation, I've found `dotnet watch` CLI to be faster than Hot Reload in Visual Studio (Blazor Server). 

I am not sure what's the differences between them?

**Comment by marinasundstrom** [comment]
From what I recently heard, they are not exactly the same. Watch watches for files to change, then recompiles, and restarts the app. "Hot reload" is the reload mechanism.

In Visual Studio there is no such watch functionality, at least not the same.

My guess is that Dotnet Watch and VS both integrate with the same basic Hot Reload functionality in different ways.

**Reply by KennethHoff** [reply]
They are both implementations of the `Edit and Continue` support; VS presumably has some built-in "trigger EnC for these files", while `dotnet watch` presumably does similarly, but as a CLI tool.

**Comment by nicholasyin** [comment]
Hot reload is not stable at all. Sometimes it appears and sometimes it doesn't without any reason you can think of.
